src/base/ossimBlockStreamBuffer.cpp

```diff
@@ -0,0 +1,270 @@
+#include <ossim/base/ossimBlockStreamBuffer.h>
+#include <ossim/base/ossimTrace.h>
+
+static ossimTrace traceDebug("BlockStreamBuffer:debug");
+
+ossim::BlockStreamBuffer::BlockStreamBuffer(ossim::istream* adaptStream, ossim_uint64 blockSize)
+:  m_adaptStream(adaptStream),
+   m_currentPosValue(0)
+{
+   m_blockInfo.m_blockSize = blockSize;
+   if(m_blockInfo.m_blockSize)
+   {
+      m_blockBuffer.resize(m_blockInfo.m_blockSize);
+      m_blockInfo.m_blockBufferPtr = &m_blockBuffer.front();
+
+//      setbuf(m_blockInfo.m_blockBufferPtr, m_blockInfo.m_blockSize);
+   }
+   setg(0, 0, 0);
+}
+
+std::streambuf* ossim::BlockStreamBuffer::setbuf (char* s, 
+                                             std::streamsize n)
+{
+   if(n>0)
+   {
+      m_blockBuffer.clear();
+      m_blockInfo.setBuffer(s,n);
+      setg(0, 0, 0);
+   }
+   return this;
+}
+
+void ossim::BlockStreamBuffer::setgPtrs()
+{
+   if(m_blockInfo.isLoaded())
+   {
+      setg(m_blockInfo.getBuffer(),
+           m_blockInfo.getBuffer() + (m_currentPosValue - m_blockInfo.m_startByte),
+           m_blockInfo.getBuffer() + m_blockInfo.m_validSize);
+   }
+   else
+   {
+      setg(0,0,0);
+   }
+
+}
+
+void ossim::BlockStreamBuffer::loadBlock()
+{
+   m_blockInfo.m_blockLoaded = false;
+   m_blockInfo.m_validSize = 0;
+   if(m_adaptStream)
+   {
+      if(m_currentPosValue < 0) m_currentPosValue = 0;
+      ossim_int64 blockIndex = m_blockInfo.getBlockIndex(m_currentPosValue);
+      m_blockInfo.m_startByte = blockIndex*m_blockInfo.m_blockSize;
+      if(!m_adaptStream->good()) m_adaptStream->clear();
+      if(m_blockInfo.m_startByte != m_adaptStream->tellg())
+      {
+         m_adaptStream->seekg(m_blockInfo.m_startByte);
+      }
+      m_adaptStream->read(m_blockInfo.m_blockBufferPtr, 
+                          m_blockInfo.m_blockSize);
+      ossim_int64 bytesRead = m_adaptStream->gcount();
+      if(!m_adaptStream->bad()&&(bytesRead>0))
+      {
+         m_blockInfo.m_blockLoaded = true;
+         m_blockInfo.m_validSize = bytesRead;
+         if((m_blockInfo.m_validSize<=0)||(!m_blockInfo.isWithinValidWindow(m_currentPosValue)))
+         {
+            m_blockInfo.m_blockLoaded = false;
+         }
+      }
+   }
+   setgPtrs();
+}
+
+std::streambuf::pos_type ossim::BlockStreamBuffer::seekoff(off_type offset, 
+                         std::ios_base::seekdir dir,
+                         std::ios_base::openmode mode)
+{
+   // make sure we are in synch with current pos
+   // gptr can be updated by other means
+   syncCurrentPosition();
+   pos_type result = pos_type(off_type(-1));
+   if(m_adaptStream)
+   {
+      // result = m_adaptStream->rdbuf()->pubseekoff(offset, dir, mode);
```

Comment:
will this commented out code be used at some point?

Reply:
removed the code and will do a pull